desktop/src/com/frostwire/gui/bittorrent/HttpDownload.java

```diff
@@ -466,15 +464,14 @@ public void onHeaders(HttpClient httpClient, Map<String, List<String>> headerFie
             //try figuring out file size from HTTP headers depending on the response.
             if (size < 0) {
                 String responseCodeStr = headerFields.get(null).get(0);
-                
-                if (responseCodeStr.contains(String.valueOf(HttpURLConnection.HTTP_OK))) {
-                    if (headerFields.containsKey("Content-Length")) {
-                        try {
-                            size = Long.valueOf(headerFields.get("Content-Length").get(0));
-                        } catch (Exception e) {}
-                    }
-                } 
-            } 
+
+                if (responseCodeStr.contains(String.valueOf(HttpURLConnection.HTTP_OK))
+                    && headerFields.containsKey("Content-Length")) {
+                    try {
+                        size = Long.valueOf(headerFields.get("Content-Length").get(0));
+                    } catch (Exception e) {}
+                }
+            }
```

Comment:
here it doesn't make much sense.
There could be other fixes on other header fields. If I accepted this fix, then I'd have to undo this if in the future, because I'd want to do things only when the response code is HTTP_OK.

See why it's like this?